my-demo-project/calc/src/main/java/calc/Calculator.java

```diff
@@ -3,17 +3,35 @@
 import java.util.List;
 
 public final class Calculator {
+    /**
+     * сделала Calculator синглтоном, подумала не имеет смысл делать много экземпляров класса
+     */
+    private static volatile Calculator instance;
 
-    private Calculator() {
+private Calculator(){
 
```

Comment:
Статические анализаторы кода. такие, как SonarQube ожидают, что методов и конструкторов с пустым телом быть не должно. Поэтому туда вставляют комментарий, объясняющий, почему тут пусто.
```suggestion
    //Blank private constructor for utility class
```